### ISSUE TITLE
Is it possible to specify a Hook based on absent key?

### ISSUE THREAD

**Issue by redruin1** [keleshev/schema#274]
Suppose I have some data with a key that is desired but not necessary:
```python
some_data = {
    "required_key_1": [ ... ],
    "required_key_2": [ ... ],
    "optional_data": "expected and desired, but not absolutely needed"
}
```
I'd like to have it setup where if `optional_data` is absent, it calls a logging function outputting a warning (i.e. "Expected 'optional_data'; are you sure your data is valid?"). Of course, `optional_data` is not required and can be inferred if needed  using `Optional`.

I'd like to setup a `Hook` to output this warning if the key is *not* present in the data, but Hook only seems to have a callback for if the key is detected. Is there any way to get this functionality? Something like:
```python
schema = Schema({
    "required_key_1": list,
    "required_key_2": list,
    Expected("optional_data"): str # or Expected(Optional("optional_data"))
})
```

**Comment by HorridModz** [comment]
You might want to try doing this outside of the schema:

```py
if not optional_data:
    warnings.warn("Expected optional_data")
# Run your schema
```

I'm not suggesting that this feature shouldn't be added to `schema`, though - a feature like `Expected("optional_data")` would definitely be helpful.
